stdlib/http/src/main/ballerina/src/http/client_endpoint.bal

```diff
@@ -243,6 +255,7 @@ public type ClientConfiguration record {|
     OutboundAuthConfig? auth = ();
     CircuitBreakerConfig? circuitBreaker = ();
     RetryConfig? retryConfig = ();
+    CookieConfig? cookieConfig = {};
```

Comment:
Why are we making this an optional type if `{}` is set as the default? Is `()` ever set as the value for this field?